feat: Language and localization support #25 (Korean)

CarBotfan opened the pull request:
- Added i18next and its required plugins(HttpBackend, LanguageDetector) to handle Language files.
- Added Language Selection UI
- Utilized i18nextBrowserLanguageDetector to automatically apply the user's default browser language upon initial visit.
- User-facing strings are now extracted into separate file for each language.
- Localized date, time, number formats on Uptime.
- Added Korean Translations for each component.

Minor issues on displaying uptime since it takes 4 seconds to refresh, which causes slight delay on changing the uptime's time format.

CarBotfan committed:
feat: Language and localization support (#25)(korean)

CarBotfan committed:
feat: Language and localization support (#25)(korean)

CarBotfan committed:
feat: Language and localization support (#25) (korean)

CarBotfan committed:
feat: Language and localization support (#25) (korean)

srini-abhiram commented on web/index.html:
```diff
@@ -1,47 +1,57 @@
 <!DOCTYPE html>
 <html lang="en">
```
The lang attribute is hardcoded to "en". It would be better to update this attribute dynamically when the language changes. You can do this in `i18next.on('languageChanged', ...)` callback

srini-abhiram commented on web/index.html:
```diff
@@ -81,21 +91,21 @@ <h2>Usage per core</h2>
             </div>
             <div id="pinned-process-container">
-                <h2>Pinned Processes</h2>
+                <h2 id="pinned_process"></h2>
```
The title for the "Pinned Processes" section is not being translated. You are missing `data-i18n="pinned_process" `attribute.

srini-abhiram commented on web/script.js:
```diff
@@ -3,26 +3,58 @@ let sortColumn = 'cpu';
+        backend: {
+            loadPath: '/locales/{{lng}}.json'
+        },
+        debug: true
```
Debug mode to be disabled.

srini-abhiram commented on web/index.html:
```diff
@@ -1,47 +1,57 @@
+            <div class="language-selector-container">
+                <select id="language-selector">
+                    <option value="" selected disabled class="placeholder">Language</option>
+                    <option value="en">English</option>
```
On page load, "Language" option is selected, but rather the actual language option should be selected. A minor UI fix can be to default to English.

srini-abhiram reviewed (changes requested):
On a different note, the language selector can be moved right next to Glances-Go title. In a future update, we could add a flag when starting the application to use a particular language as default. Everything else looks solid!

CarBotfan committed:
feat: Language and localization support (#25) (korean)

CarBotfan commented:
- Added i18next.on callback, now lang attribute is updated dynamically when language changes. 
- Fixed mistype where Pinned process doesn't have a data-i18n attribute
- i18n Debug has been disabled
- Slightly moved the language selection menu's position
- Language menu now shows current language options as shown below
<img width="1261" height="914" alt="image" src="https://github.com/user-attachments/assets/24ff005b-97c1-444c-90f3-291e718437c7" />

Thanks for the review!

srini-abhiram reviewed (changes requested):
The default selected language is now empty, please default it to english, other changes look solid

CarBotfan committed:
feat: Language and localization support (#25) (korean)

CarBotfan commented:
Fixed it. 
The problem was i18next.language returning English's code in the form of 'en-US'.

srini-abhiram commented:
Thanks for contributing! Give it a star if it helped you.
